### PR TITLE
introduce 'slice::slice::read{u,i}{8,16,32,64}{be,le}()' API

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

**Comment by aep** [comment]
makes sense to have, but not happy about the naming

read implies you can read multiple packed integers, but actually this is a conversion function and it ignores endianess.

i'd rather call it to_u8

**Comment by aep** [comment]
actually that tells me MutSlice is poorly named.
while MutSlice is a writer with a write state, Slice does not have a read state

**Comment by aep** [comment]
oh wait, completely missed the offset.

i guess you CAN read multiple packed integers that way, so the name is fine

**Comment by jwerle** [comment]
@aep Yeah I am rewriting to use the `byteorder` module and make `to_u{8,16,32,64}{be,le}()` variants. I can rename `to_` back to `read`

**Comment by aep** [comment]
host endianess is fine, push8 does that too. 

i'm just using memcpy in push because shift is so confusing. But if you can verify your code is always using host endianess, its fine

**Comment by jwerle** [comment]
@aep just pushed up some changes if you are free to take a look

File: modules/slice/src/slice.zz [hunk 1]
```diff
@@ -156,7 +157,177 @@ export fn sub(Slice *self, usize from, usize mut size) -> Slice
...
+}
 
+/// read slice as an i8 little endian value
+export fn readi8le(Slice *self, usize offset = 0) -> i8
+    where len(self->mem) >= 0
+    where offset < len(self->mem)
+{
+    let ptr = self->mem + offset;
+    u8 val = ptr[0];
+    return as<i8>(val | (val & 128) * 0x1fffffe);
```
Comment (aep) [inline modules/slice/src/slice.zz:267]: how does this work?

Comment (jwerle) [reply modules/slice/src/slice.zz:267]: oops this was left over before a `:w!` in my editor. This shouldn't be there

**Review by aep — APPROVED** [review]
lgtm

**Pull request merged**